include/revng/tests/unit/ModelType.inc

```diff
@@ -32,40 +32,43 @@ static bool verify(const model::Binary &Tree, bool Assert);
 static bool checkSerialization(const TupleTree<model::Binary> &Tree);
 
 BOOST_AUTO_TEST_CASE(PrimitiveTypes) {
-  revng_check(verify(*PrimitiveType::makeVoid(), {}, true));
-  revng_check(verify(*PrimitiveType::makeUnsigned(1), {}, true));
-  revng_check(verify(*PrimitiveType::makeUnsigned(2), {}, true));
-  revng_check(verify(*PrimitiveType::makeUnsigned(4), {}, true));
-  revng_check(verify(*PrimitiveType::makeUnsigned(8), {}, true));
-  revng_check(verify(*PrimitiveType::makeUnsigned(16), {}, true));
-  revng_check(verify(*PrimitiveType::makeSigned(1), {}, true));
-  revng_check(verify(*PrimitiveType::makeSigned(2), {}, true));
-  revng_check(verify(*PrimitiveType::makeSigned(4), {}, true));
-  revng_check(verify(*PrimitiveType::makeSigned(8), {}, true));
-  revng_check(verify(*PrimitiveType::makeSigned(16), {}, true));
-  revng_check(verify(*PrimitiveType::makeFloat(2), {}, true));
-  revng_check(verify(*PrimitiveType::makeFloat(4), {}, true));
-  revng_check(verify(*PrimitiveType::makeFloat(8), {}, true));
-  revng_check(verify(*PrimitiveType::makeFloat(10), {}, true));
-  revng_check(verify(*PrimitiveType::makeFloat(12), {}, true));
-  revng_check(verify(*PrimitiveType::makeFloat(16), {}, true));
-  revng_check(verify(*PrimitiveType::makeGeneric(1), {}, true));
-  revng_check(verify(*PrimitiveType::makeGeneric(2), {}, true));
-  revng_check(verify(*PrimitiveType::makeGeneric(4), {}, true));
-  revng_check(verify(*PrimitiveType::makeGeneric(8), {}, true));
-  revng_check(verify(*PrimitiveType::makeGeneric(10), {}, true));
-  revng_check(verify(*PrimitiveType::makeGeneric(12), {}, true));
-  revng_check(verify(*PrimitiveType::makeGeneric(16), {}, true));
-  revng_check(verify(*PrimitiveType::makeNumber(1), {}, true));
-  revng_check(verify(*PrimitiveType::makeNumber(2), {}, true));
-  revng_check(verify(*PrimitiveType::makeNumber(4), {}, true));
-  revng_check(verify(*PrimitiveType::makeNumber(8), {}, true));
-  revng_check(verify(*PrimitiveType::makeNumber(16), {}, true));
-  revng_check(verify(*PrimitiveType::makePointerOrNumber(1), {}, true));
-  revng_check(verify(*PrimitiveType::makePointerOrNumber(2), {}, true));
-  revng_check(verify(*PrimitiveType::makePointerOrNumber(4), {}, true));
-  revng_check(verify(*PrimitiveType::makePointerOrNumber(8), {}, true));
-  revng_check(verify(*PrimitiveType::makePointerOrNumber(16), {}, true));
+  TupleTree<model::Binary> T;
+  T->Architecture() = model::Architecture::x86_64;
+
+  revng_check(verify(*PrimitiveType::makeVoid(), *T, true));
```

Comment:
> tests: ensure architecture is always available 

Why?

Reply:
Because we need to know which set of helper names to check against. Hence every `model::Binary` now always has to have an architecture field set.

Reply:
Now that you say it... Yeah, this seems like a left-over from an earlier iteration.

I'll try dropping this and see if anything pops up. Good catch!